test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/VerifyCommandTests.cs

```diff
@@ -280,5 +282,611 @@ public async Task VerifyCommand_VerifyOnPackageSignedWithoutAllowedCertificateFa
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("true", true)]
+        [InlineData("true", false)]
+        [InlineData("TRUE", true)]
+        [InlineData("TRUE", false)]
+        public async Task Verify_RepositorySignedPackageWithUntrustedCertificate_RepositoryTag_AllowUntrustedRootSetTrue_Succeeds(string trust, bool fingerPrintOption)
+        {
+            // Arrange
+            var nupkg = new SimpleTestPackageContext("A", "1.0.0");
+
+            using (var pathContext = new SimpleTestPathContext())
+            using (var zipStream = await nupkg.CreateAsStreamAsync())
+            using (var testCertificate = new X509Certificate2(_testFixture.UntrustedSelfIssuedCertificateInCertificateStore))
+            {
+                var packagePath = Path.Combine(pathContext.WorkingDirectory, nupkg.PackageName);
+
+                zipStream.Seek(offset: 0, loc: SeekOrigin.Begin);
+
+                using (var fileStream = File.OpenWrite(packagePath))
+                {
+                    zipStream.CopyTo(fileStream);
+                }
+
+                //Act
+                string certFingerprint = SignatureTestUtility.GetFingerprint(testCertificate, HashAlgorithmName.SHA256);
+                string repoServiceIndex = "https://serviceindex.test/v3/index.json";
+                string signedPackagePath = await SignedArchiveTestUtility.RepositorySignPackageAsync(testCertificate, packagePath, pathContext.PackageSource, new Uri(repoServiceIndex));
+
+                // Arrange
+                var certificateFingerprintString = SignatureTestUtility.GetFingerprint(testCertificate, HashAlgorithmName.SHA256);
+
+                var trustedSignersSectionContent = $@"
+    <trustedSigners>
+        <repository name=""MyCert"" serviceIndex = ""{repoServiceIndex}"">
+            <certificate fingerprint=""{certificateFingerprintString}"" hashAlgorithm=""SHA256"" allowUntrustedRoot=""{trust}"" />
+        </repository>
+    </trustedSigners>
+";
+                SimpleTestSettingsContext.AddSectionIntoNuGetConfig(pathContext.WorkingDirectory, trustedSignersSectionContent, "configuration");
+                string fingerprint = fingerPrintOption ? $"-CertificateFingerprint {certificateFingerprintString};abc;def" : string.Empty;
+
+                // Act
+                var verifyResult = CommandRunner.Run(
+                    _nugetExePath,
+                    pathContext.PackageSource,
+                    $"verify {signedPackagePath} -Signatures {fingerprint}",
+                    waitForExit: true);
+
+                // Assert
+                // If allowUntrustedRoot is set true in nuget.config then verify succeeds for cert with untrusted root.
+                verifyResult.Success.Should().BeTrue();
+                verifyResult.AllOutput.Should().Contain(_noTimestamperWarningCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("true", true)]
```

Comment:
Do we need all the caps tests?
Feels like they're not adding much as long we've tested the parsing once, we should good, correct?

That's also testing something that should be tested in the configuration reading, since that is a functionality specific to the config reading.

Reply:
Sure, removed.